Plutarch/Internal/PlutusType.hs

```diff
@@ -140,9 +141,13 @@ class PCon a where
   pcon :: a s -> Term s a
 
 class PMatch a where
-  -- | Pattern match over Plutarch Terms via a Haskell datatype
+  -- | Pattern match over Plutarch Terms via a Haskell datatype. All branches are strictly evaluated, use `pmatchSum` if you need them lazy.
   pmatch :: Term s a -> (a s -> Term s b) -> Term s b
 
+-- | A less efficient delayed 'pmatch' for use with partial pattern matches in @do@ notation.
+pmatchSum :: PlutusType a => Term s a -> (a s -> Term s b) -> Term s b
+pmatchSum x f = PI.pforce (pmatch x (PI.pdelay . f))
```

Comment:
The issue is that it's easy to accidentally use `pmatch` and have a non-obvious bug.